WordPress/gutenberg
# Tests: Jest - Add typeahead plugin

![Screen Capture on 2020-06-22 at 11-21-12](https://user-images.githubusercontent.com/2322354/85305566-39af7a80-b47b-11ea-96d2-3a5cb1a58a70.gif)

This update adds a [type ahead](https://github.com/jest-community/jest-watch-typeahead) plugin for Jest. This feature use to be part of Jest core. However, it was abstracted out a couple of years ago. It's a tiny interaction that adds a lot of value for dev experience in regards to testing.


## How has this been tested?

* run `npm install`
* run `npm run test-unit:watch`
* press `p` to pause the tests
* press `p` again to go into "Pattern Mode"
* start typing
* check out the typeahead matching
* press up/down arrows to go through the list
* press enter to run the test

## Checklist:
- [n/a] My code follows the WordPress code style. 
- [n/a] My code follows the accessibility standards. 
- [n/a] My code has proper inline documentation. 
- [n/a] I've included developer documentation if appropriate. 
- [n/a] I've updated all React Native files affected by any refactorings/renamings in this PR.
